src/treetensornetworks/projttns/projttnsum.jl

```diff
@@ -1,17 +1,25 @@
 """
 ProjTTNSum
 """
-struct ProjTTNSum{V}
-  pm::Vector{ProjTTN{V}}
-  function ProjTTNSum(pm::Vector{ProjTTN{V}}) where {V}
-    return new{V}(pm)
+struct ProjTTNSum{T<:AbstractProjTTN}
+  pm::Vector{T} where {T}
+  function ProjTTNSum(pm::Vector{T}) where {T}
+    return new{T}(pm)
   end
 end
+#ToDo: define accessor functions
 
 copy(P::ProjTTNSum) = ProjTTNSum(copy.(P.pm))
 
+# The following constructors don't generalize well, maybe require to pass AbstractProjTTN instead of TTN and remove these? 
 ProjTTNSum(ttnos::Vector{<:TTN}) = ProjTTNSum([ProjTTN(M) for M in ttnos])
+function ProjTTNSum(init_states::Vector{<:TTN}, ttnos::Vector{<:TTN})
```

Comment:
It makes more sense to me to just use the constructor `ProjTTNSum(terms::Vector{<:AbstractProjTTN})` and pass it `Vector{ProjTTNApply}` instead of defining a specialized constructor.

Reply:
Specifically, two sets of `Vector{<:TTN}` could really mean anything, _a priori_ I wouldn't expect this to construct a sum of `ProjTTNApply` as opposed to a sum of some other type of `AbstractProjTTN`.